pkg/monitor/intervalcreation/pod.go

```diff
@@ -403,13 +458,13 @@ func buildTransitionsForCategory(locatorToConditions map[string][]monitorapi.Eve
 }
 
 func sanitizeTime(nextInterval monitorapi.EventInterval, startTime, endTime time.Time) monitorapi.EventInterval {
-	if nextInterval.To.After(endTime) {
+	if !endTime.IsZero() && nextInterval.To.After(endTime) {
```

Comment:
I don't know if this line is related, but, when I run this, I'm getting the `To` set to the current date.  So, when I look at a chart from 4/29, it thinks the `To` is now and the bars go all the way to the right making the other bars look very short.

Reply:
It's because some pods aren't observed to end and those have null end dates which render long durations on the intervals chart.